### DISCUSSION TITLE
Context-aware settings

### DISCUSSION THREAD

**Discussion by ylorn** [wojciech-kulik/xcodebuild.nvim#123] [Q&A] [answered]
Hi, thanks for this awesome plugin! I followed the tutorial https://wojciechkulik.pl/ios/the-complete-guide-to-ios-macos-development-in-neovim and it is been very pleasant to use it.

I am fairly new to Neovim(using LazyVim distro right now), so I am wondering if there is way to make the `xcodebuild.nvim` settings be aware of the context? Like to make the keymap only work in a Xcode project/workspace or Swift Package folder? So that `whickkey` menu doesn't gets polluted in a non-apple-development environment.

Thanks!

**Comment by wojciech-kulik** [comment, marked as answer]
Hi! After `setup` call you can check: 

```lua
if require("xcodebuild.project.config").is_project_configured() then
  -- setup bindings
end
```

It should work :)

**Reply by wojciech-kulik** [reply]
Alternatively, you can setup local bindings only for Swift buffers using autocommands

**Reply by wojciech-kulik** [reply]
@ylorn does it answer your question?

**Reply by ylorn** [reply]
Yes, marked it as answered 🙏🏼
